src/NServiceBus.Persistence.CosmosDB.TransactionalSession/CosmosTransactionalSession.cs

```diff
@@ -0,0 +1,16 @@
+namespace NServiceBus.TransactionalSession
+{
+    using Persistence.CosmosDB;
+
+    sealed class CosmosTransactionalSession : TransactionalSession
+    {
+        public CosmosTransactionalSession()
+        {
+            Defaults(s =>
+            {
+                s.GetOrCreate<TransactionInformationConfiguration>().ExtractPartitionKeyFromHeaders(new ControlMessagePartitionKeyExtractor());
```

Comment:
a create at this point would maybe be too late for the transport to take the configuration into account?

Looking at https://github.com/Particular/NServiceBus.Persistence.CosmosDB/blob/1.1.1/src/NServiceBus.Persistence.CosmosDB/Transaction/Transaction.cs#L9, do we need to ensure that the `Transaction` feature runs afterwards somehow? (e.g. when there is no class from the settings it creates a new one locally that isn't pushed back into the settings)

Reply:
That's exactly what I mean. We need to change the way the Transaction feature deals with those settings

Reply:
I think that is a necessary change because otherwise we might run into random feature ordering issues which might break the configuration.

Reply:
How about https://github.com/Particular/NServiceBus.Persistence.CosmosDB/pull/427